create_derivatives/helpers.py

```diff
@@ -22,8 +20,8 @@ def matching_files(directory, prefix=None, suffix=None,
     Returns:
         files (lst): a list of files that matched the identifier.
     """
-    files = sorted([f for f in listdir(directory) if (
-        isfile(join(directory, f)) and not f.startswith((".", "Thumbs")))])
+    files = sorted([f for f in Path.iterdir(directory) if (
+        Path.is_file(Path(directory).joinpath(f))) and not f.startswith((".", "Thumbs")))])
```

Comment:
```suggestion
        Path(directory, f).is_file() and not f.startswith((".", "Thumbs")))])
```
(you may want to take a closer look at the docs)